assets/compat/config_spec.yaml

```diff
@@ -82,6 +82,32 @@ max-chan-size:
   range: "[1,1000000000]"
   integral: true
   units: satoshis
+tor:
+  type: object
+  name: Tor Config
+  nullable: false
+  spec:
+    use-tor-only:
+      type: boolean
+      name: Use Tor for all traffic
+      description: >-
+        Use the tor proxy even for connections that are reachable on clearnet. This will hide your node's public IP
+        address, but will slow down your node's performance
+      nullable: false
+      default: false
+    stream-isolation:
+      type: boolean
+      name: Stream Isolation
+      description: >-
+        Enable Tor stream isolation by randomizing user credentials for each
+        connection. With this mode active, each connection will use a new circuit.
+        This means that multiple applications (other than lnd) using Tor won't be mixed
+        in with lnd's traffic.
+
+        This option may not be used when 'Use Tor for all traffic' is disabled, since direct
+        connections compromise source IP privacy by default.
+      nullable: false
+      default: false
```

Comment:
Is the slowdown significant in tor-only mode?

Reply:
imo, yes. But it really depends on how you're defining "significant". My opinion is that things should never use tor unless the user opts in or it is necessary. Tor is not used in our system as a privacy enhancement primarily. Further, connecting to .onion's always uses tor. This is only a consideration for clearnet connections. The tor network has also been historically unreliable, and reliability is far and away a much more important consideration in Lightning.

Reply:
I'm mostly just wanting to know whether, when you tried it, you noticed any issues

Reply:
I haven't tried it yet, actually. I should.